test/user_class.cpp

```diff
@@ -537,3 +537,33 @@ TEST(MSGPACK_USER_DEFINED, test_non_intrusive)
 
     EXPECT_EQ(t1.name(), t2.name());
 }
+
+struct nvp_base {
+    int a;
+    int b;
+    MSGPACK_DEFINE_MAP(MSGPACK_NVP("aaa", a), b);
+};
+
+struct nvp_derived : nvp_base {
+    int c;
+    std::string d;
+    MSGPACK_DEFINE_MAP(MSGPACK_NVP("ccc", c), MSGPACK_NVP("base", MSGPACK_BASE(nvp_base)), MSGPACK_NVP("ddd", d));
+};
+
+TEST(MSGPACK_NVP, combination)
+{
+    msgpack::sbuffer sbuf;
+    nvp_derived d1;
+    d1.a = 1;
+    d1.b = 2;
+    d1.c = 3;
+    d1.d = "ABC";
+
+    msgpack::pack(sbuf, d1);
+    msgpack::object_handle oh = msgpack::unpack(sbuf.data(), sbuf.size());
```

Comment:
Good point. If the converting logic have a bug that doing something wrong at packing and converting processes, it could be generate a right result.

I will update the test.